askup/views.py

```diff
@@ -229,19 +234,29 @@ def get_queryset(self):
 @login_required
 def user_profile_view(request, user_id):
     """Provide the user profile view."""
-    user = get_object_or_404(User, pk=user_id)
+    profile_user = get_object_or_404(User, pk=user_id)
     return render(
         request,
         'askup/user_profile.html',
         {
-            'first_name': user.first_name,
-            'last_name': user.last_name,
-            'email': user.email,
-            'questions_count': get_user_questions_count(user.id),
-            'answers_count': get_user_answers_count(user.id),
-            'own_score': get_user_score_by_id(user.id),
-            'is_owner': user.id == request.user.id,
-            'is_student': check_user_has_groups(request.user, 'student')
+            'first_name': profile_user.first_name,
+            'last_name': profile_user.last_name,
+            'email': profile_user.email,
+            'own_profile': int(user_id) == request.user.id,
+            'own_score': get_user_score_by_id(profile_user.id),
+            'is_owner': profile_user.id == request.user.id,
+            'is_student': check_user_has_groups(request.user, 'student'),
+            'own_correct_answers': get_user_correct_answers_count(profile_user.id),
```

Comment:
Why not use `user_id` ?

Reply:
That's right, I can. But I'll need to cast it into an int type, cause `user_id` is getting here as a string.